### ISSUE TITLE
Misleading message for SC2002

### ISSUE THREAD

**Issue by gtmtech** [koalaman/shellcheck#2608]
#### For bugs

- Rule Id (if any, e.g. SC1000): SC2002
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

The problem comes from the remedy suggestion to:

```sh
cat a.txt | while read -r line ; do echo $line ; done

SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```

The syntax suggested seems to suggest rather than `cat | while` - you do while < file | ...` which is wrong. 

You need to do `while....; done < file` - I wonder if the message could be a bit clearer about this as it seems to tell you to fix it incorrectly.

```sh
cat a.txt 
one
two
three

$ while read -r line < a.txt; do echo $line ; done
one
one
one

$ while read -r line; do echo $line ; done < a.txt
one
two
three
```

#### Here's what shellcheck currently says:

```
SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```

#### Here's what I wanted or expected to see:

```
SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead or 'while condition; do .... ; done < file` if a looping construct.
```

**Comment by abpicoli1** [comment]
Also I would like to add that the SC2002 rule doesn't quite apply for commands that have different behavior when receiving input from stdin x a file provide in the arguments.

For example `cat FILE | wc -l` will return only the number of lines in a file.
`wc -l "$FILE"` will return an output containing the number of lines _plus the filename_. 

Which means that either I use `cat FILE | wc -l` or use `wc -l "$FILE" | awk ' { print $1 }'` . No benefit at all.
